src/main/java/frc/robot/subsystems/Elevator.java

```diff
@@ -0,0 +1,265 @@
+import edu.wpi.first.wpilibj.smartdashboard.SmartDashboard;
+import edu.wpi.first.wpilibj.util.Color;
+import edu.wpi.first.wpilibj.util.Color8Bit;
+import edu.wpi.first.wpilibj2.command.Command;
+import edu.wpi.first.wpilibj2.command.CommandScheduler;
+import edu.wpi.first.wpilibj2.command.SubsystemBase;
+import edu.wpi.first.wpilibj2.command.button.Trigger;
+import frc.robot.Constants;
+import frc.robot.Constants.ElevatorConstants;
+import frc.robot.Constants.MotorIDConstants;
+import frc.robot.Robot;
+
+public class Elevator extends SubsystemBase {
+  private TalonFX elevatorMotor1;
+  private TalonFX elevatorMotor2;
+  private CurrentLimitsConfigs currentLimit = new CurrentLimitsConfigs();
+  private MotorOutputConfigs invertMotor =
+      new MotorOutputConfigs().withInverted(InvertedValue.Clockwise_Positive);
+  private static Mechanism2d mech = new Mechanism2d(2, 2);
+  private static ComplexWidget widg;
+  private DigitalInput elvLimitSwitch;
+  private MechanismLigament2d elevatorMech;
+  private MotionMagicConfigs elvMotionMagic =
+      new MotionMagicConfigs()
+          .withMotionMagicCruiseVelocity(MMVel)
+          .withMotionMagicAcceleration(MMVel * 2)
+          .withMotionMagicJerk(MMVel * 20);
+
+  public static final SoftwareLimitSwitchConfigs elvSoftLimit =
+      new SoftwareLimitSwitchConfigs()
+          .withForwardSoftLimitEnable(true)
+          .withForwardSoftLimitThreshold(heightToRotations(Constants.ElevatorConstants.kTopLimit))
+          .withReverseSoftLimitEnable(true)
+          .withReverseSoftLimitThreshold(
+              heightToRotations(Constants.ElevatorConstants.kBottomLimit));
+  public static final Slot0Configs loopCfg =
+      new Slot0Configs()
+          .withKP(Constants.ElevatorConstants.P)
+          .withKI(Constants.ElevatorConstants.I)
+          .withKD(Constants.ElevatorConstants.D);
+  private ElevatorSim m_elevatorSim;
+
+  public Elevator() {
+    // TODO: set up for canivore
+    currentLimit.StatorCurrentLimit = 90;
+    currentLimit.SupplyCurrentLimit = 70;
+    currentLimit.SupplyCurrentLowerLimit = 40;
+    currentLimit.SupplyCurrentLowerTime = 1;
+    currentLimit.StatorCurrentLimitEnable = true;
+    currentLimit.SupplyCurrentLimitEnable = true;
+    elevatorMotor1 = new TalonFX(MotorIDConstants.kElevatorMotor1, Constants.RIOName);
+    elevatorMotor2 = new TalonFX(MotorIDConstants.kElevatorMotor2, Constants.RIOName);
+    elevatorMotor1.getConfigurator().apply(loopCfg);
+    elevatorMotor1.getConfigurator().apply(elvSoftLimit);
+    elevatorMotor1.getConfigurator().apply(currentLimit);
+    elevatorMotor1.getConfigurator().apply(invertMotor);
+    elevatorMotor1.getConfigurator().apply(elvMotionMagic);
+    elevatorMotor2.setControl(new Follower(MotorIDConstants.kElevatorMotor1, true));
+    elvLimitSwitch = new DigitalInput(Constants.ElevatorConstants.elvLimitID);
+    new Trigger(CommandScheduler.getInstance().getActiveButtonLoop(), elvLimitSwitch::get)
```

Comment:
```suggestion
    new Trigger(elvLimitSwitch::get)
```
I believe the Trigger gets added to the active button loop by default.